BankApp/BankLibrary/Bank.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+
+namespace BankLibrary
+{
+    public class Bank<T> : IBank<T> where T : Account
+    {
+        private readonly List<T> _accounts = new();
+
+        public void OpenAccount(OpenAccountParameters parameters)
+        {
+            CheckValidTypeAccount(parameters);
+            T newAccount = null;
+            CreateAccount(parameters.AccountCreated, () => parameters.Type == AccountType.Deposit
+            ? newAccount = new DepositAccount(parameters.Amount) as T
+            : newAccount = new OnDemandAccount(parameters.Amount) as T);
+
+            AddingNotification(parameters, newAccount);
```

Comment:
that way sounds better
```suggestion
            SubscribeOnNotifications(parameters, newAccount);
```